### PR TITLE
#97140-ContentTypeOptional

### PR DESCRIPTION
**Pull request by kulbhushan-skf**
97140-ContentTypeOptional - Need to test defect #21 in User services. So made the changes to request.go file.

### CODE REVIEW

File: api/godog/requests.go [hunk 1]
```diff
@@ -136,9 +137,10 @@ func (api *BaseFeature) ExecuteTheRequestWithPayload(payload []byte) (err error)
 		return errors.Wrapf(err, "http.NewRequest failed - Payload: `%s`", string(payload))
 	}
 
-	req.Header = api.Request.Headers
-	req.Header.Set("Content-Type", "application/json")
-
+	if payload != nil {
+		req.Header = api.Request.Headers
```
Comment (rickardenglund) [inline api/godog/requests.go:141]: This one should always be set

**Comment by kulbhushan-skf** [comment]
@rickardenglund Please find the PR.

File: api/godog/requests.go [hunk 2]
```diff
@@ -136,9 +137,10 @@ func (api *BaseFeature) ExecuteTheRequestWithPayload(payload []byte) (err error)
...
+		req.Header.Set("Content-Type", "application/json")
```
Comment (rickardenglund) [inline api/godog/requests.go:142]: And I think that we should remove this completly.

if it is needed it should be specified in the gherkin spec.

**Review by rickardenglund — CHANGES_REQUESTED** [review]

**Comment by kulbhushan-skf** [comment]
@rickardenglund updated comments and tested with users repo.

**Review by rickardenglund — APPROVED** [review]

**Pull request merged**